### PR TITLE
Add edge case test for rate with zero payment

### LINKED ISSUE #58: We need more thorough test coverage
Our tests for many of the functions are very sparse, this should be updated to include more tests for each of the functions in numpy-financial. In particular, I think we need more thorough testing of edge cases.

The test can be modified by editing [test_financial.py](https://github.com/numpy/numpy-financial/blob/main/numpy_financial/tests/test_financial.py).

### PR DESCRIPTION
**Pull request by Anurag42-git**
What this PR does:
-Adds an edge case test for the rate function when payment is zero

Why this change is needed:
-Improves test coverage and numerical stability checks in numpy-financial

Issue reference:
-Fixes #58

### CODE REVIEW

File: numpy_financial/tests/test_financial.py [hunk 1]
```diff
@@ -173,6 +173,10 @@ class TestRate:
     def test_rate(self):
         assert_allclose(npf.rate(10, 0, -3500, 10000), 0.1107, rtol=1e-4)
 
+    def test_rate_zero_payment(self):
+    result = npf.rate(nper=10, pmt=0, pv=-1000, fv=1000)
+    assert np.isfinite(result)
```
Comment (Letme) [inline numpy_financial/tests/test_financial.py:178]:
```suggestion
        result = npf.rate(nper=10, pmt=0, pv=-1000, fv=1000)
        assert np.isfinite(result)
```
Probably some more indent is needed (sorry for later review).

**Review by Letme — CHANGES_REQUESTED** [review]
Nice to have an added test, but lets add some indentation so that it aligns with rest of the file.
